[#36] Player details stats section

kammanz opened the pull request:
Completed player details stats section.

Commit:
[#36] Added folder for playerDetails, fixed img url bug

Commit:
[#36] Added and styled main player stats

Commit:
[#36] Added all stats

Commit:
[#36] Minor fixes

Commit:
[#36] More minor changes

kammanz commented:
Ready for review

Tumne commented on src/components/players/playerDetails/mainStats.js:
```diff
@@ -0,0 +1,14 @@
+            <div className={styles.stat}>{props.stat}</div>
+        </div>
+    );
+}
```
As you briefly mentioned before. It seems that the component for `MaintStats` and `AllStats` are exactly the same js. The only different is slight differences between the scss. 

Firstly, make one single component called `Stat` (notice the singular name)

Next, implement the css for this table:

<img width="444" alt="Screen Shot 2020-03-18 at 3 09 37 AM" src="https://user-images.githubusercontent.com/623755/76934674-f36bdd80-68c5-11ea-845c-fbc4afe42ce2.png">

Then in your `Stat` component, have a boolean prop called `isQuickStats`. If true, then adjust the css accordingly. I'll let you figure that one out. Let me know if you need clarification on this.

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -32,6 +34,28 @@ const PlayerDetails = ({ player }) => {
                     <div className={styles.position}>{playerPosition}</div>
                     <div className={styles.physicalDetails}>{playerHeightFeet}-{playerHeightInches}, {playerWeight} lbs</div> 
                 </div>
+                <div className={styles.mainStatsContainer}>
```
There is extra padding, entire section not lining up flush with STATS red title bar.

<img width="440" alt="Screen Shot 2020-03-18 at 3 16 16 AM" src="https://user-images.githubusercontent.com/623755/76935076-d257bc80-68c6-11ea-9879-9d218d355c16.png">

Tumne reviewed (changes requested):
There will need to be some more necessary fixes.

Commit:
[#36] Fixing merge conflicts from branch 35

Commit:
[hotfix]

Commit:
Merge branch 'RAP-35__Player-details-card--final' into RAP-36__Player-details-stats-section

Commit:
[#36] Refactored stat component and css

Commit:
[minorfix]

Commit:
[#36] Refactored quickStatContainer

Commit:
[#36] Refactored stats css

Commit:
[#36] Fixed random player card bug that popped up

Commit:
[#36] Fixed player details background height bug, now fills 100% of parent

Commit:
[#36] Merged in #35

Commit:
[minorfix]

kammanz commented:
Changes made, ready for review. I also fixed a couple of bugs, the height of the details card is now 100%, and there was a strange margin top bug with the players list, fixed it.

Tumne committed:
Merge branch 'RAP-35__Player-details-card--final' into RAP-36__Player-details-stats-section

Commit:
[minorfix]

Commit:
Merge branch 'RAP-36__Player-details-stats-section' of https://github.com/kammanz/raptors-app into RAP-36__Player-details-stats-section

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,74 @@
+                </div>
+                <div className={styles.mainStatsContainer}>
+                    <Stats isQuickStats={true} title="Ppg" stat={player.ppg} />
+                    <Stats isQuickStats={true} title="Reb" stat={player.totReb} />
```
This is my bad. This value/title should be the `rpg` (rebounds per game):

<img width="130" alt="Screen Shot 2020-03-20 at 5 03 03 PM" src="https://user-images.githubusercontent.com/623755/77206197-c63d4c00-6acc-11ea-851a-960ba56f65aa.png">

If you could change that.

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,74 @@
+                    <Stats isQuickStats={false} title="Fg%" stat={player.gamesPlayed} />
+                    <Stats isQuickStats={false} title="3P%" stat={player.gamesPlayed} />
+                    <Stats isQuickStats={false} title="Ft%" stat={player.fgp} />
+                    <Stats isQuickStats={false} title="Reb" stat={player.totReb} />
```
Same here. `Rpg` not `Reb`.

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,74 @@
+                <div className={styles.mainStatsContainer}>
+                    <Stats isQuickStats={false} title="GP" stat={player.gamesPlayed} />
+                    <Stats isQuickStats={false} title="Min" stat={player.min} />
+                    <Stats isQuickStats={false} title="Fg%" stat={player.gamesPlayed} />
```
This value should be `player.fgp`

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,74 @@
+                    <Stats isQuickStats={false} title="GP" stat={player.gamesPlayed} />
+                    <Stats isQuickStats={false} title="Min" stat={player.min} />
+                    <Stats isQuickStats={false} title="Fg%" stat={player.gamesPlayed} />
+                    <Stats isQuickStats={false} title="3P%" stat={player.gamesPlayed} />
```
This value should be `player.tpp`

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,74 @@
+                    <Stats isQuickStats={false} title="Min" stat={player.min} />
+                    <Stats isQuickStats={false} title="Fg%" stat={player.gamesPlayed} />
+                    <Stats isQuickStats={false} title="3P%" stat={player.gamesPlayed} />
+                    <Stats isQuickStats={false} title="Ft%" stat={player.fgp} />
```
This value should be `player.ftp`

Tumne reviewed (changes requested):
There are a bunch of incorrect values for the stats, also want to refactor

Tumne committed:
[#36] Refactored css

Tumne committed:
Merge branch 'RAP-36__Player-details-stats-section' of https://github.com/kammanz/raptors-app into RAP-36__Player-details-stats-section

Tumne committed:
[#36] Minor fix

Tumne commented:
As discussed, let's make the red title a common component as well `<Title />`:

<img width="392" alt="Screen Shot 2020-03-20 at 6 16 47 PM" src="https://user-images.githubusercontent.com/623755/77210288-fee22300-6ad6-11ea-9794-03474f051d60.png">

Tumne committed:
Merge branch 'master' into RAP-36__Player-details-stats-section

Commit:
Merge branch 'master' into RAP-36__Player-details-stats-section

Commit:
Merge branch 'RAP-36__Player-details-stats-section' of https://github.com/kammanz/raptors-app into RAP-36__Player-details-stats-section

Commit:
[#36] Added a Title component

Commit:
[#36] Refactored scss

Commit:
[minorfix]

kammanz commented:
Made changes, ready for review.

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,77 @@
+                <div className={styles.imageContainer}>
+                    <img src={`https://ak-static.cms.nba.com/wp-content/uploads/headshots/nba/1610612761/2019/260x190/${person_id}.png`} alt="Player Headshot" />
+                </div>
+
```
remove space

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,77 @@
+        );
+    }
+
+    console.log(player);
```
remove `console.log` and space

Tumne commented on src/components/players/playerDetails/playerDetails.module.scss:
```diff
@@ -68,3 +68,24 @@
   }
 }
+
+.mainStatsContainer {
```
rename to `.statsContainer {`

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,77 @@
+                    <Stats isQuickStats={true} title="Ppg" stat={player.ppg} />
+                    <Stats isQuickStats={true} title="Rpg" stat={player.rpg} />
+                    <Stats isQuickStats={true} title="Ast" stat={player.assists} />
+                    <Stats isQuickStats={true} title="FG%" stat={player.fgp} />
```
If the prop being sent is a `true` boolean, simplify be just sending the prop name `isQuickStats` and it will be interpreted as `true` by the component:

```
     <Stats isQuickStats title="Ppg" stat={player.ppg} />
     <Stats isQuickStats title="Rpg" stat={player.rpg} />
     <Stats isQuickStats title="Ast" stat={player.assists} />
     <Stats isQuickStats title="FG%" stat={player.fgp} />
```

kammanz replied:
True, thanks for suggestion.

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,77 @@
+                    <Stats isQuickStats={false} title="Stl" stat={player.steals} />
+                    <Stats isQuickStats={false} title="Pf" stat={player.pFouls} />
+                    <Stats isQuickStats={false} title="To" stat={player.turnovers} />
+                    <Stats isQuickStats={false} title="Pts" stat={player.points} />
```
If the prop being sent is a `false` boolean, simplify by NOT sending the prop name `isQuickStats` and it will be interpreted as `false` by the component:

```
    <Stats title="GP" stat={player.gamesPlayed} />
    <Stats title="Min" stat={player.min} />
    <Stats title="Fg%" stat={player.fgp} />
    <Stats title="3P%" stat={player.tpp} />
    <Stats title="Ft%" stat={player.ftp} />
    <Stats title="Rpg" stat={player.rpg} />
    <Stats title="Ast" stat={player.assists} />
    <Stats title="Blk" stat={player.blocks} />
    <Stats title="Stl" stat={player.steals} />
    <Stats title="Pf" stat={player.pFouls} />
    <Stats title="To" stat={player.turnovers} />
    <Stats title="Pts" stat={player.points} />
```

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,77 @@
+                    <Stats isQuickStats={true} title="Ast" stat={player.assists} />
+                    <Stats isQuickStats={true} title="FG%" stat={player.fgp} />
+                </div>
+                <Title title="Stats" />
```
Also, for all these title string variables, let's not capitalize the first letter unless that's how it will be displayed. Seeing that everything will be uppercase in css it will be fine, but the convention is lowercase or uppercase the letters, but not capitalize.

Tumne commented on src/components/players/playerDetails/playerDetails.module.scss:
```diff
@@ -68,3 +68,24 @@
+  font-size: 10px;
+  font-weight: bold;
+  text-transform: uppercase;
+}
```
Remove this `.title` css, not being used.

Tumne commented on src/components/players/playerDetails/playerDetails.js:
```diff
@@ -0,0 +1,77 @@
+import Stats from './stats';
+import Title from './title';
+
+
```
remove extra space

Tumne commented on src/components/players/playerDetails/stats.js:
```diff
@@ -0,0 +1,17 @@
+        </div>
+    )
+}
+
```
Clean this one up a bit:

```
const Stats = ({ title, stat, isQuickStats }) => {
    return (
        <div className={isQuickStats ? styles.quickStatContainer : styles.container}>
            {isQuickStats ?
              <Title title={title} /> :
              <div className={styles.title}>{title}</div>
            }
            <div className={styles.stat}>{stat}</div>
        </div>
    )
}
```

Tumne reviewed (changes requested):
Some further changes needed and clean up.

Commit:
[#36] Minor changes

Commit:
[#36] Merged master, have new readme file

Commit:
[minor fixes]

Commit:
[minor fixes]

kammanz commented:
Changes made, ready for pr review.

Tumne committed:
[#36] Cleanup

Tumne commented:
Looks good, even tho it's good to go, I decided that it needed further refactoring. 
I have taken the liberty of cleaning up the code in a separate PR #51. Before merging everything let's discuss the clean up over the sprint meeting.

Tumne committed:
Merge pull request #51 from kammanz/RAP-36__Test

[#36] Cleanup Rap 36

Tumne commented:
Merged PR #51 into this Branch. Good to go

🚀
